clients/fossid-webapp/src/test/kotlin/FossIdClientReturnTypeTest.kt

```diff
@@ -80,85 +71,109 @@ class FossIdClientReturnTypeTest : StringSpec({
 
     "Scans for project can be listed when there is none" {
         service.listScansForProject("", "", PROJECT_CODE_1) shouldNotBeNull {
-            checkResponse("list scans")
-            toList<Scan>() should beEmpty()
+            checkResponse("list scans", true)
+            data shouldNotBeNull {
+                values should beEmpty()
+            }
         }
     }
 
     "Scans for project can be listed when there is some" {
         service.listScansForProject("", "", PROJECT_CODE_2) shouldNotBeNull {
-            checkResponse("list scans")
-            toList<Scan>() shouldNot beEmpty()
+            checkResponse("list scans", true)
+            data shouldNotBeNull {
+                values shouldNot beEmpty()
+            }
         }
     }
 
     "Scan results can be listed when there is none" {
         service.listScanResults("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list scan results")
-            toList<FossIdScanResult>() should beEmpty()
+            data shouldNotBeNull {
+                values should beEmpty()
+            }
         }
     }
 
     "Scan results can be listed when there is some" {
         service.listScanResults("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list scan results")
-            toList<FossIdScanResult>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                values shouldNot beEmpty()
+            }
         }
     }
 
     "Identified files can be listed when there is none" {
         service.listIdentifiedFiles("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list identified files")
-            toList<IdentifiedFile>() should beEmpty()
+            data shouldNotBeNull {
+                values should beEmpty()
+            }
         }
     }
 
     "Identified files can be listed when there is some" {
         service.listIdentifiedFiles("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list identified files")
-            toList<IdentifiedFile>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                values shouldNot beEmpty()
+            }
         }
     }
 
     "Marked as identified  files can be listed when there is none" {
         service.listMarkedAsIdentifiedFiles("", "", SCAN_CODE_1) shouldNotBeNull {
             checkResponse("list marked as identified files")
-            toList<MarkedAsIdentifiedFile>() should beEmpty()
+            data shouldNotBeNull {
+                values should beEmpty()
+            }
         }
     }
 
     "Marked as identified  files can be listed when there is some" {
         service.listMarkedAsIdentifiedFiles("", "", SCAN_CODE_2) shouldNotBeNull {
             checkResponse("list marked as identified files")
-            toList<MarkedAsIdentifiedFile>() shouldNot beEmpty()
+            data shouldNotBeNull {
+                values shouldNot beEmpty()
```

Comment:
Hint: To avoid a name clash of `beEmpty()` for collections (see below) and maps (here), you could use

    import io.kotest.matchers.maps.beEmpty as beEmptyMap

and then

    this should beEmptyMap()

here.

Reply:
Yes ! This was the bit I was missing and the reason I checked only the values.